Touch enhancement and Install/Uninstall script

encrustace opened the pull request:
- [x] Install Script added
- [x]  Uninstall Script added
- [x]  Touch and cursor movement accuracy improved
- [x]  Autoselect layout in install script

encrust-ace committed:
Daemon needed to restart after laptop reboot (fixed)

encrustace committed:
Merge branch 'iamkroot:master' into master

encrust-ace committed:
install script

encrust-ace committed:
uninstall script

encrust-ace committed:
Number touch and just cursor move is more accurate

encrustace commented:
@iamkroot  Please review and test, for me it is now touches are very accurate

encrust-ace committed:
Now install script will ask to user to select numpad model

encrust-ace committed:
changed install path

iamkroot commented on src/main.rs:
```diff
@@ -351,13 +357,6 @@ impl Numpad {
                     Some(key) => {
                         self.grab();
                         self.state.finger_state = FingerState::Touching;
```
Removing this code seems wrong. Now we can't hold a key to repeat it. 

> By default, when you hold down a key on your keyboard, the letter or symbol will be repeated until you release the key

(Gnome docs)

iamkroot commented:
Regarding the install script- I have an aversion to reading/writing shell scripts. Would much rather prefer a saner language such as python. (maybe you can get gpt to do it :P)

encrust-ace committed:
1. Install script now select layout automatically\n2.Installation folder is now usr/local/bin

encrust-ace committed:
Reverted the code for continue press

encrustace commented:
Hey @iamkroot  I have reverted the continue press code.
I have added autoselect layout in install script.
I choose bash to avoid python dependency.

A suggestion if you can integegrate layout auto selection in build itself, that would make it more easy to setup.

iamkroot commented on install.sh:
```diff
@@ -0,0 +1,35 @@
+    echo "Config file already exists!"
+    cat /etc/xdg/asus_numpad.toml
+else
+    keyboardModel=$(cat /sys/class/dmi/id/board_name)
```
This doesn't work if it is something other than one of the known models. It is also possible that a laptop has the same layout as one already supported - so it can reuse the same name. We should present a list of layouts to the user in case it doesn't belong to the ones we know of.

iamkroot commented on src/main.rs:
```diff
@@ -325,10 +325,16 @@ impl Numpad {
                 debug!("Keyup {:?}", key);
 
                 if self.layout.needs_multikey(key) {
+if self.state.pos.dist_sq(self.state.tap_start_pos) <= Self::TAP_JITTER_DIST {
```
Could we please put this change in a separate PR?

iamkroot commented on src/main.rs:
```diff
@@ -513,7 +518,6 @@ impl Numpad {
 
 fn main() -> Result<()> {
     env_logger::init();
-
```
Unnecessary. Please revert all whitespace changes

iamkroot commented on src/main.rs:
```diff
@@ -325,10 +325,16 @@ impl Numpad {
                     self.dummy_kb.multi_keyup(&self.layout.multikeys(key));
+}
                 } else {
+if self.state.pos.dist_sq(self.state.tap_start_pos) <= Self::TAP_JITTER_DIST {
```
Please ensure the file is properly formatted with `cargo fmt`.

iamkroot reviewed (changes requested):
As I see it, there's three changes in this PR:
1. Install/uninstall scripts
2. Jitter
3. UX325EA model name

Would be better to split them up into separate PRs so they are easier to review.
